### PR TITLE
MAGECLOUD-3467: Rewrite CronTest to use Cloud Docker

### PR DESCRIPTION
**Pull request by BaDos**
### Description

CronTest was rewrote using Cloud Docker

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3467

### Manual testing scenarios

None

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: tests/functional/_data/modules/Magento/CronTest/Model/TestJob.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
```
Comment (oshmyheliuk) [inline tests/functional/_data/modules/Magento/CronTest/Model/TestJob.php]: add declare(strict_types=1) or remove emty line

File: tests/functional/Codeception/MagentoDb.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        $this->disconnectDatabases();
+    }
+
+    /**
+     * This method is overridden to avoid connection attempts before running Docker
+     * {@inheritdoc}
+     */
+    public function _initialize()
+    {}
```
Comment (oshmyheliuk) [inline tests/functional/Codeception/MagentoDb.php]: This syntax looks strange I think `public function _initialize() {}` will be better.

File: tests/functional/Robo/Tasks/CopyToDocker.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+     * @param string $source
+     * @param string $destination
+     * @param string $container
+     */
+    public function __construct(string $source, string $destination, string $container)
+    {
+        $this->container = $container;
+        $this->source = $source;
+        $this->destination = $destination;
+    }
```
Comment (bbatsche) [inline tests/functional/Robo/Tasks/CopyToDocker.php:76]: Not a blocker really, but Robo convention is usually a more fluent syntax. Something along the lines of:

```php
$this->copyToDocker($src)
    ->container($containerName)
    ->destination($somePath)
    ->run()
```

This structure of putting everything in the constructor works fine, but is definitely different that the style I've seen elsewhere

Comment (bbatsche) [reply tests/functional/Robo/Tasks/CopyToDocker.php:76]: Or looking at the approach we've done else where, starting with the `$container` in the constructor and `source()` and `destination()` as methods.

File: tests/functional/Codeception/Docker.php [hunk 4]
```diff
@@ -213,18 +250,47 @@ public function grabFileContent(string $source, string $container = self::DEPLOY
...
+     * @param array $variables
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function runBinMagentoCommand(string $command, string $container, array $variables = []): bool
+    {
+        return $this->taskBash($container)
+            ->envVars($this->prepareVariables($variables))
+            ->exec(sprintf('php %s/bin/magento %s', $this->_getConfig('system_magento_dir'), $command))
+            ->run()
```
Comment (bbatsche) [inline tests/functional/Codeception/Docker.php:279]: On the Magecat backlog, I want to add `->printOutput(false)` to just about all the Robo tasks so they don't blow up the console as much. If we need to log or assert against the output of those tasks, we can use `->getOutputData()` after the task has been run.

It'd probably be nice to head this off in ECE Tools so we don't run into the same problem here.

**Review by bbatsche — DISMISSED** [review]
Nothing blocking the merge, but I think there are some pieces that could be improved, noted above.

File: tests/functional/_data/modules/Magento/CronTest/Model/TestJob.php (hunk 1 above)
Comment (BaDos) [reply tests/functional/_data/modules/Magento/CronTest/Model/TestJob.php]: Done

File: tests/functional/Codeception/MagentoDb.php (hunk 2 above)
Comment (BaDos) [reply tests/functional/Codeception/MagentoDb.php]: Done

File: tests/functional/Robo/Tasks/CopyToDocker.php (hunk 3 above)
Comment (BaDos) [reply tests/functional/Robo/Tasks/CopyToDocker.php:76]: Done

File: tests/functional/Codeception/Docker.php (hunk 4 above)
Comment (BaDos) [reply tests/functional/Codeception/Docker.php:279]: I will do it in the scope of my next task

**Review by oshmyheliuk — APPROVED** [review]

File: src/Docker/Compose/IntegrationV2Compose.php [hunk 5]
```diff
@@ -138,8 +138,7 @@ public function build(Repository $repository): array
             'cron',
             false,
             ['db'],
```
Comment (shiftedreality) [inline src/Docker/Compose/IntegrationV2Compose.php:140]: Remove cron service completely

**Review by shiftedreality — APPROVED** [review]

File: src/Docker/Compose/IntegrationV2Compose.php (hunk 5 above)
Comment (BaDos) [reply src/Docker/Compose/IntegrationV2Compose.php:140]: I will do it in the scope of my next task

**Pull request merged**